test/erc20.spec.ts

```diff
@@ -1,128 +1,252 @@
-    it('should revert when the owner account does not have enough balance', async () => {
-      const sender = account1
-      const owner = account2
-      const recipient = account3
-      const amount = 2500000
-
-      await expect(
-        ERC20.connect(sender).transferFrom(
-          await owner.getAddress(),
-          await recipient.getAddress(),
-          amount
+    describe('on L2', () => {
+      it('should should have a name', async () => {
+        expect(await ERC20_L2.name()).to.equal(name)
+      })
+
+      it('should have a total supply equal to the initial supply', async () => {
+        expect(await ERC20_L2.totalSupply()).to.equal(initialSupplyL2)
+      })
+
+      it("should give the initial supply to the creator's address", async () => {
+        expect(await ERC20_L2.balanceOf(await walletL2.getAddress())).to.equal(
+          initialSupplyL2
         )
-      ).to.be.revertedWith(
-        "Can't transfer from the desired account because it doesn't have enough balance."
-      )
+      })
     })
 
-    it('should revert when the sender does not have enough of an allowance', async () => {
-      const sender = account2
-      const owner = account1
-      const recipient = account3
-      const amount = 2500000
+    describe('`transfer` on L1', () => {
+      it('should revert when the sender does not have enough balance', async () => {
+        const sender = walletL1
+        const recipient = walletL1_2
+        const amount = initialSupplyL1.add(ethers.BigNumber.from('2500000'))
 
-      await expect(
-        ERC20.connect(sender).transferFrom(
-          await owner.getAddress(),
-          await recipient.getAddress(),
+
+        await expect(
+          ERC20_L1.connect(sender).transfer(await recipient.getAddress(), amount)
+        ).to.be.revertedWith(
+          "You don't have enough balance to make this transfer!"
+        )
+      })
+
+      it('should succeed when the sender has enough balance', async () => {
+        const sender = walletL1
+        const recipient = walletL1_2
+        const amount = ethers.BigNumber.from('2500000')
+
+        await ERC20_L1.connect(sender).transfer(await recipient.getAddress(), amount)
+
+        expect(await ERC20_L1.balanceOf(await walletL1.getAddress())).to.equal(
+          initialSupplyL1.sub(amount)
+        )
+
+        expect(await ERC20_L1.balanceOf(await walletL1_2.getAddress())).to.equal(
           amount
         )
-      ).to.be.revertedWith(
-        "Can't transfer from the desired account because you don't have enough of an allowance."
-      )
+      })
     })
 
-    it('should succeed when the owner has enough balance and the sender has a large enough allowance', async () => {
-      const sender = account2
-      const owner = account1
-      const recipient = account3
-      const amount = 2500000
+    describe('`transfer` on L2', () => {
+      it('should revert when the sender does not have enough balance', async () => {
```

Comment:
This is the only test that fails unexpectedly.
The call to `transfer()` does not revert when it should.
As a result, this test also causes the subsequent test to fail, but otherwise, all tests except this test work as expected.

Currently, I'm not sure why the transaction is not reverting.